src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -3492,15 +3497,38 @@ private ConversionOperatorDeclarationSyntax TryParseConversionOperatorDeclaratio
                     ? this.EatToken()
                     : this.EatToken(SyntaxKind.ExplicitKeyword);
 
-                if (this.CurrentToken.ContextualKind == SyntaxKind.ExtensionKeyword)
+                var possiblyExtension = this.CurrentToken.ContextualKind == SyntaxKind.ExtensionKeyword;
+                if (possiblyExtension)
                 {
-                    this.Reset(ref point);
-                    return null;
+                    // If we have `implicit extension E` then this is definitely an extension type as it could not be parsed
+                    // as anything else.
+                    if (IsTrueIdentifier(this.PeekToken(1)))
+                    {
+                        this.Reset(ref point);
+                        return null;
+                    }
+
+                    // we have `implicit extension ...` technically this could be something like `implicit
+                    // extension.operator X(` (an implicit conversion with a explicit interface impl name). For compat,
+                    // determine the parsing strategy based on the lang version.
+                    if (IsCompatBreakingFeatureEnabled(MessageID.IDS_FeatureExtensions))
```

Comment:
>if (IsCompatBreakingFeatureEnabled(MessageID.IDS_FeatureExtensions)) [](http://example.com/codeflow?start=20&length=68)

Why not look at the next token though? I think `implicit extension.operator X(` should be parsed as an operator regardless of language version. I would prefer us to explicitly target scenarios that we explicitly decided to parse as type declarations rather than changing parsing based on language version across the board

Reply:
Done.